src/target_entry.rs

```diff
@@ -0,0 +1,104 @@
+use ffi;
+use glib;
+use glib::translate::*;
+use gobject_ffi;
+use std::mem;
+use TargetFlags;
+
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct TargetEntry {
+    pub target: String,
+    pub flags: TargetFlags,
+    pub info: u32,
+}
+
+impl TargetEntry {
+    pub fn new(target: &str, flags: TargetFlags, info: u32) -> TargetEntry {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gtk_target_entry_new(target.to_glib_none().0, flags.bits(), info))
+        }
+    }
+}
+
+#[doc(hidden)]
+impl Uninitialized for TargetEntry {
+    #[inline]
+    unsafe fn uninitialized() -> Self {
+        mem::uninitialized()
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtr<'a, *const ffi::GtkTargetEntry> for TargetEntry {
+    type Storage = &'a Self;
+
+    #[inline]
+    fn to_glib_none(&'a self) -> Stash<'a, *const ffi::GtkTargetEntry, Self> {
+        let ptr: *const TargetEntry = &*self;
+        Stash(ptr as *const ffi::GtkTargetEntry, self)
```

Comment:
You need create new GtkTargetEntry and fill it fields and assign it to first stash parameter
As we need hold both Self and this then `Stash<'a, *const ffi::GtkTargetEntry, (Self, ffi::GtkTargetEntry)>`